terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -1233,15 +1233,15 @@ type tfContextLogger struct {
 	ctx context.Context // bind context to struct to satisfy interface
 }
 
-func (l *tfContextLogger) Debugf(format string, args ...any) {
+func (l *tfContextLogger) Debug(format string, args ...any) {
```

Comment:
I think we should keep these at `Debugf, Infof, ...`, as they do not use slog and perform formatting.

Reply:
While that is true, there are multiple places where we use `slog.Logger` as a `tfContextLogger` and since `slog` doesn't implement any formatting functions we would have to wrap it, which I think would be unintended? Alternatively, I could convert the functions to non-formatting functions since you mentioned that not using formatting with `slog` would be preferred anyways.

Reply:
> there are multiple places where we use `slog.Logger` as a `tfContextLogger`

`tfContextLogger` is not an interface. What do you mean by this?

Reply:
Im sorry, this was badly formulated from my side. `tfContextLogger` has to implement `debugLog` which `slog.Logger` needs to implement as well for it to be used in parts of the code. Since `slog.Logger` has no formatting functions I had to rewrite the `debugLog` interface which in turn caused me to rewrite `tfContextLogger`.